pkg/loadbalancer/experimental/bpf_reconciler.go

```diff
@@ -511,10 +511,18 @@ func (ops *BPFOps) updateFrontend(fe *Frontend) error {
 	ip := fe.Address.AddrCluster.AsNetIP()
 	if fe.Address.IsIPv6() {
 		svcKey = lbmap.NewService6Key(ip, fe.Address.Port, u8proto.ANY, fe.Address.Scope, 0)
-		svcVal = &lbmap.Service6Value{}
+		if ops.cfg.PerServiceLoadBalancerAlgo {
+			svcVal = &lbmap.Service6ExtendedValue{}
+		} else {
+			svcVal = &lbmap.Service6Value{}
+		}
 	} else {
 		svcKey = lbmap.NewService4Key(ip, fe.Address.Port, u8proto.ANY, fe.Address.Scope, 0)
-		svcVal = &lbmap.Service4Value{}
+		if ops.cfg.PerServiceLoadBalancerAlgo {
+			svcVal = &lbmap.Service4ExtendedValue{}
```

Comment:
We seem to be missing the implementation for setting the algorithm here?
It shouldn't be fairly easy to do the equal changes to the new load-balancer code:
- service.go should include the algorithm field
- reflector.go should fill it in when ingesting the serivce
- bpf_reconciler.go should set the algorithm in the value
- add the necessary code to DumpLBMaps to show the algorithm

And I would also add a test case for it. `testdata/clusterip.txtar` might be a good starting point. Just add in the annotation and check that BPF value gets correctly set.

Reply:
I will move implementation in experimental package outside of this PR, sorry I overlooked it.